src/Algebra/Properties/Semiring/Binomial.agda

```diff
@@ -0,0 +1,289 @@
+-- Next, a lemma which is independent of commutativity
+
+  module _ {n} (i : Fin (suc n)) where
+
+    open Binomial n using (term)
+
+    x*lemma : x * term i ≈ (n C toℕ i) × Binomial.binomial (suc n) (suc i)
+    x*lemma = begin
+      x * term i                                  ≡⟨⟩
+      x * ((n C k) × (x ^ k * y ^ (n ∸ k)))       ≈˘⟨ *-congˡ (×-assoc-* (n C k) _ _) ⟩
+      x * ((n C k) × x ^ k * y ^ (n ∸ k))         ≈˘⟨ *-assoc x ((n C k) × x ^ k) _ ⟩
+      (x * (n C k) × x ^ k) * y ^ (n ∸ k)         ≈⟨ *-congʳ (×-comm-* (n C k) _ _) ⟩
+      (n C k) × x ^ (suc k) * y ^ (n ∸ k)         ≡⟨⟩
+      (n C k) × x ^ (suc k) * y ^ (suc n ∸ suc k) ≈⟨ ×-assoc-* (n C k) _ _ ⟩
+      (n C k) × Binomial.binomial (suc n) (suc i) ∎
+      where
+        open ≈-Reasoning
+        k = toℕ i
+
+
+------------------------------------------------------------------------
+-- Next, a lemma which does require commutativity
+
+  module _ (x*y≈y*x : x * y ≈ y * x) where
+
+    module _ (n : ℕ) where
+
+      open Binomial n using (binomial; term; term₁; term₂)
+
+      y*lemma : (j : Fin n) → let i = inject₁ j in
+        y * term (suc j) ≈ (n C toℕ (suc j)) × Binomial.binomial (suc n) (suc i)
+      y*lemma j = begin
+        y * term (suc j)
+          ≈⟨ ×-comm-* nC[j+1] y (binomial (suc j)) ⟩
+        nC[j+1] × (y * binomial (suc j))
+          ≈⟨ ×-congʳ nC[j+1] (y*x^m*y^n≈x^m*y^[n+1] x*y≈y*x [j+1] [n-j-1]) ⟩
+        nC[j+1] × (x ^ [j+1] * y ^ [n-j])
+          ≈˘⟨ ×-congʳ nC[j+1] (*-congʳ (^-congʳ x [k+1]≡[j+1])) ⟩
+        nC[j+1] × (x ^ [k+1] * y ^ [n-j])
+          ≈˘⟨ ×-congʳ nC[j+1] (*-congˡ (^-congʳ y [n-k]≡[n-j])) ⟩
+        nC[j+1] × (x ^ [k+1] * y ^ [n-k])
+          ≡⟨⟩
+        nC[j+1] × (x ^ [k+1] * y ^ [n+1]-[k+1])
+          ≡⟨⟩
+        (n C toℕ (suc j)) × Binomial.binomial (suc n) (suc i) ∎
+        where
+          i           = inject₁ j
+          k           = toℕ i
+          k≡j         : k ≡ toℕ j
+          k≡j         = toℕ-inject₁ j
+
+          [k+1]       = ℕ.suc k
+          [j+1]       = toℕ (suc j)
+          [n-k]       = n ∸ k
+          [n+1]-[k+1] = [n-k]
+          [n-j-1]     = n ∸ [j+1]
+          [n-j]       = ℕ.suc [n-j-1]
+          nC[j+1]     = n C [j+1]
+
+          [k+1]≡[j+1] : [k+1] ≡ [j+1]
+          [k+1]≡[j+1] = cong suc k≡j
+
+          [n-k]≡[n-j] : [n-k] ≡ [n-j]
+          [n-k]≡[n-j] = begin
+            [n-k]     ≡⟨ cong (n ∸_) k≡j ⟩
+            n ∸ toℕ j ≡⟨ +-∸-assoc 1 (toℕ<n j) ⟩
+            [n-j]     ∎ where open ≡-Reasoning
+          open ≈-Reasoning
+
+------------------------------------------------------------------------
+-- Now, a lemma characterising the sum of the term₁ and term₂ expressions
+
+      open ≈-Reasoning
+
+      private
+        n<ᵇ1+n : (n <ᵇ suc n) ≡ true
+        n<ᵇ1+n with n <ᵇ suc n in eq with <⇒<ᵇ (n<1+n n)
+        ... | true | p rewrite eq = ≡-refl
+
+      lemma₁₊₂₌ₑₓₚ : ∀ i → term₁ i + term₂ i ≈ Binomial.term (suc n) i
```

Comment:
This name is both not particularly informative and pretty damn hard to read (and type). Maybe a different name is in order?

Reply:
I think that works